src/sonic-db-cli/sonic-db-cli.cpp

```diff
@@ -0,0 +1,239 @@
+#include <execution>
+#include <iostream>
+
+#include <swss/dbconnector.h>
+#include <swss/sonicv2connector.h>
+
+#include "sonic-db-cli.h"
+
+static std::string name_space;
+static std::string db_or_operation;
+const char* empty_str = "";
+
+void printUsage(const po::options_description &all_options)
+{
+    std::cout << "usage: sonic-db-cli [-h] [-s] [-n NAMESPACE] db_or_op [cmd ...]" << std::endl;
+    std::cout << all_options << std::endl;
+    
+    std::cout << "**sudo** needed for commands accesing a different namespace [-n], or using unixsocket connection [-s]" << std::endl;
+    std::cout << "" << std::endl;
+    std::cout << "Example 1: sonic-db-cli -n asic0 CONFIG_DB keys \\*" << std::endl;
+    std::cout << "Example 2: sonic-db-cli -n asic2 APPL_DB HGETALL VLAN_TABLE:Vlan10" << std::endl;
+    std::cout << "Example 3: sonic-db-cli APPL_DB HGET VLAN_TABLE:Vlan10 mtu" << std::endl;
+    std::cout << "Example 4: sonic-db-cli -n asic3 APPL_DB EVAL \"return {KEYS[1],KEYS[2],ARGV[1],ARGV[2]}\" 2 k1 k2 v1 v2" << std::endl;
+    std::cout << "Example 5: sonic-db-cli PING | sonic-db-cli -s PING" << std::endl;
+    std::cout << "Example 6: sonic-db-cli SAVE | sonic-db-cli -s SAVE" << std::endl;
+    std::cout << "Example 7: sonic-db-cli FLUSHALL | sonic-db-cli -s FLUSHALL" << std::endl;
+}
+
+void printRedisReply(redisReply* reply)
+{
+    switch(reply->type)
+    {
+    case REDIS_REPLY_INTEGER:
+        std::cout << reply->integer;
```

Comment:
>std::cout [](http://example.com/codeflow?start=8&length=9)

Is printf faster?

Reply:
cout is little bit faster than printf: https://stackoverflow.com/questions/896654/cout-or-printf-which-of-the-two-has-a-faster-execution-speed-c